app/src/main/java/fuzion24/device/vulnerability/vulnerabilities/kernel/CVE_2016_5195.java

```diff
@@ -0,0 +1,102 @@
+package fuzion24.device.vulnerability.vulnerabilities.kernel;
+
+import android.content.Context;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import fuzion24.device.vulnerability.util.CPUArch;
+import fuzion24.device.vulnerability.vulnerabilities.VulnerabilityTest;
+
+/**
+ * Created by Till Busse on 21/11/16.
+ * Java implementation to test dirtyCow vulnerability.
+ */
+
+public class CVE_2016_5195 implements VulnerabilityTest {
+    static {
+        System.loadLibrary("dirtyCow");
+    }
+    private final String file1Name = "file1";
+    private final String file2Name = "file2";
+    private final String file1Content = "This is File one.";
+    private final String file2Content = "Mo0h";
+    @Override
+    public String getCVEorID() {
+        return "CVE-2016-5195";
+    }
+
+    private native int runDirtyCow(Object [] paths);
+
+    @Override
+    public boolean isVulnerable(Context context) throws Exception {
+        try{
+
+            List<Object> paths = createFiles(context);
+            boolean isWritten = checkFileContent(context, file1Name, file1Content);
+            if (!isWritten)
+                throw new Exception("Error running test. File could not be created with specific content");
+
+            int checkVal = runDirtyCow(paths.toArray());
+            if (checkVal == 0){
+                isWritten = checkFileContent(context, file1Name, file2Content);
+                return isWritten;
+            }else {
+                throw new Exception("Error running test. Errno: " + checkVal);
+            }
+        }finally {
+            // delete files when done
+            File file;
+            file = new File(context.getFilesDir(), file1Name);
+            file.delete();
+            file = new File(context.getFilesDir(), file1Name);
```

Comment:
This should be `file2Name` I think (yeah, I know that this repo is pretty dead, but still :wink: )

Reply:
thanks =D